src/overlays/actors/ovl_Bg_Ice_Objects/z_bg_ice_objects.c

```diff
@@ -30,23 +38,209 @@ const ActorInit Bg_Ice_Objects_InitVars = {
     (ActorFunc)BgIceObjects_Update,
     (ActorFunc)BgIceObjects_Draw,
 };
-*/
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_Bg_Ice_Objects/BgIceObjects_Init.s")
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_Bg_Ice_Objects/BgIceObjects_Destroy.s")
+InitChainEntry sInitChain[] = {
+    ICHAIN_VEC3F_DIV1000(scale, 100, ICHAIN_STOP),
+};
+
+void BgIceObjects_Init(Actor* thisx, GlobalContext* globalCtx) {
+    s32 pad;
+    BgIceObjects* this = THIS;
+    ColHeader* colHeader = NULL;
+
+    Actor_ProcessInitChain(&this->dyna.actor, sInitChain);
+    DynaPolyInfo_SetActorMove(&this->dyna, 0);
+    DynaPolyInfo_Alloc(&D_060003F0, &colHeader);
+    Math_Vec3f_Copy(&this->targetPos, &this->dyna.actor.initPosRot.pos);
+    this->actionFunc = BgIceObjects_Idle;
+    this->dyna.dynaPolyId =
+        DynaPolyInfo_RegisterActor(globalCtx, &globalCtx->colCtx.dyna, &this->dyna.actor, colHeader);
+    this->dyna.actor.params = 0;
+}
+
+void BgIceObjects_Destroy(Actor* thisx, GlobalContext* globalCtx) {
+    s32 pad;
+    BgIceObjects* this = THIS;
+
+    DynaPolyInfo_Free(globalCtx, &globalCtx->colCtx.dyna, this->dyna.dynaPolyId);
+}
+
+/*
+ * Checks which of the eight possible x and z positions the block is at,
+ * defaulting to the maximum x wall or minimum z wall. Each x and z position
+ * has only one possible wall or pit on each side of it.
+ */
+void BgIceObjects_SetNextTarget(BgIceObjects* this, GlobalContext* globalCtx) {
+    static s16 sXStarts[] = {
+        -1060, -1200, -1240, -1387, -1580, -1680, -1780,
+    };
+    static s16 sZStarts[] = {
+        -580, -660, -780, -820, -860, -900, -1087,
+    };
+    static s16 sZStops[7][2] = {
+        { -580, -1087 }, { -780, -1260 }, { -340, -820 }, { -260, -1260 },
+        { -340, -860 },  { -660, -1260 }, { -340, -740 },
+    };
+    static s16 sXStops[7][2] = {
+        { -860, -1580 },  { -1240, -1780 }, { -860, -1680 }, { -860, -1680 },
+        { -1387, -1680 }, { -860, -1200 },  { -860, -1800 },
+    };
```

Comment:
kinda feeling like these are too big to be in function, unless it was needed to match. same for the other actor if that one has the same thing

Reply:
Fair. I'll just put them above that function.